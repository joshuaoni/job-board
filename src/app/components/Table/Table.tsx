"use client";

import Image from "next/image";
import { useRouter } from "next/navigation";

interface Job {
  id: string;
  job_title: string;
  job_type: string;
  required_skills: string;
  company_name: string;
  job_location_name: string;
  created_at: string;
  company_logo: string | null;
  languages: string;
  tags: string;
}

interface TableProps {
  data: Job[];
}

export const Table = ({ data = [] }: TableProps) => {
  return (
    <div className="app-table w-full my-4">
      <TableHeading />
      <TableBody jobs={data} />
    </div>
  );
};

const TableHeading = () => {
  return (
    <div className="app-table-heading w-full h-[39.292px] rounded-[7.76px] pt-[11.65px] pb-[11.65px] bg-[#D6D6D6] gap-4 grid grid-cols-[2fr_1fr_1fr_1fr_1fr] items-center mb-4">
      <TextElement text="Job Title" />
      <TextElement text="Job Type" />
      <TextElement text="Skills" />
      <TableHeadingBox element={<TextElement2 text="Languages" />} />
      <TableHeadingBox element={<TextElement2 text="Tags" />} />
    </div>
  );
};

const TextElement = ({ text }: { text: string }) => {
  return (
    <div
      // style={{ width: `${width}px` }}
      className="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
    >
      <p className="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]">
        {text}
      </p>
    </div>
  );
};

const TableHeadingBox = ({ element }: { element: React.ReactNode }) => {
  return <div className="  h-[16px] ">{element}</div>;
};

const TextElement2 = ({ text }: { text: string }) => {
  return (
    <div
      // style={{ width: `${width}px`, left: `${left}px` }}
      className="h-[16px] text-[#898989]  flex items-center"
    >
      <p className="font-urbanist font-bold text-[13.59px] leading-[100%] tracking-[5%]">
        {text}
      </p>
    </div>
  );
};

interface TableBodyProps {
  jobs: Job[];
}

const TableBody = ({ jobs }: TableBodyProps) => {
  if (!jobs.length) {
    return (
      <div className="w-full py-8 text-center text-gray-500">
        No jobs found matching your criteria
      </div>
    );
  }
  return (
    <div className="app-table-body w-full">
      {jobs.map((job) => (
        <TableBodyRow key={job.id} job={job} />
      ))}
    </div>
  );
};

interface TableBodyRowProps {
  job: Job;
}

const TableBodyRow = ({ job }: TableBodyRowProps) => {
  const router = useRouter();
  const timeAgo = getTimeAgo(new Date(job.created_at));

  const handleClick = () => {
    router.push(`/jobs/${job.id}`);
  };

  return (
    <div
      onClick={handleClick}
      className="w-full h-[51px] border-b border-white grid grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center hover:bg-gray-50 cursor-pointer transition-colors"
    >
      <TitleDetails
        company={job.company_name}
        location={job.job_location_name}
        title={job.job_title}
        timeAgo={timeAgo}
      />
      <JobType type={job.job_type} />
      <JobSkills skills={job.required_skills} />
      <LanguagesBox width="152" languages={job.languages || "Not specified"} />
      <TagsBox width="139" tags={job.tags || "Not specified"} />
    </div>
  );
};

const getTimeAgo = (date: Date): string => {
  const now = new Date();
  const diffInSeconds = Math.floor((now.getTime() - date.getTime()) / 1000);

  const intervals = {
    year: 31536000,
    month: 2592000,
    week: 604800,
    day: 86400,
    hour: 3600,
    minute: 60,
  };

  for (const [unit, seconds] of Object.entries(intervals)) {
    const interval = Math.floor(diffInSeconds / seconds);
    if (interval >= 1) {
      return `${interval} ${unit}${interval === 1 ? "" : "s"} ago`;
    }
  }

  return "Just now";
};

const JobType = ({ type }: { type: string }) => {
  return (
    <div className="w-full h-[19px]  flex items-center justify-start">
      <p className="font-inter font-normal text-[16px] leading-[100%] tracking-[5%] text-black text-center">
        {type}
      </p>
    </div>
  );
};

const JobSkills = ({ skills }: { skills: string }) => {
  const skillsList = skills.split(",").map((skill) => skill.trim());
  const displaySkill = skillsList[0];
  const remainingCount = skillsList.length - 1;

  return (
    <div className="w-full flex justify-start items-center gap-2 overflow-hidden">
      <div className="flex items-center">
        <span className="text-sm text-black truncate" title={skills}>
          {displaySkill}
        </span>
        {remainingCount > 0 && (
          <span
            className="ml-1 px-2 py-0.5 bg-gray-100 rounded-full text-xs font-medium text-gray-600"
            title={skillsList.slice(1).join(", ")}
          >
            +{remainingCount}
          </span>
        )}
      </div>
    </div>
  );
};

const LanguagesBox = ({
  width,
  languages,
}: {
  width: string;
  languages: string;
}) => {
  const languageList = languages.split(",").map((lang) => lang.trim());
  const displayLanguage = languageList[0];
  const remainingCount = languageList.length - 1;

  return (
    <div className="w-full flex justify-start items-center overflow-hidden">
      <div className="flex items-center">
        <span className="text-sm text-black truncate" title={languages}>
          {displayLanguage}
        </span>
        {remainingCount > 0 && (
          <span
            className="ml-1 px-2 py-0.5 bg-gray-100 rounded-full text-xs font-medium text-gray-600"
            title={languageList.slice(1).join(", ")}
          >
            +{remainingCount}
          </span>
        )}
      </div>
    </div>
  );
};

const TagsBox = ({ width, tags }: { width: string; tags: string }) => {
  const tagsList = tags.split(",").map((tag) => tag.trim());
  const displayTag = tagsList[0];
  const remainingCount = tagsList.length - 1;

  return (
    <div className="w-full overflow-hidden">
      <div className="flex justify-start items-center">
        <span className="text-sm text-black truncate" title={tags}>
          {displayTag}
        </span>
        {remainingCount > 0 && (
          <span
            className="ml-1 px-2 py-0.5 bg-blue-50 rounded-full text-xs font-medium text-blue-600"
            title={tagsList.slice(1).join(", ")}
          >
            +{remainingCount}
          </span>
        )}
      </div>
    </div>
  );
};

const TitleDetails = ({
  company,
  location,
  title,
  timeAgo,
}: {
  company: string;
  location: string;
  title: string;
  timeAgo: string;
}) => {
  return (
    <div className="app-title-details h-[51px] rounded-[10px] p-[4px] pl-0 gap-[10px] flex items-center">
      <GappedContainer
        company={company}
        location={location}
        title={title}
        timeAgo={timeAgo}
      />
    </div>
  );
};

const GappedContainer = ({
  company,
  location,
  title,
  timeAgo,
}: {
  company: string;
  location: string;
  title: string;
  timeAgo: string;
}) => {
  return (
    <div className="w-full h-full bg-[#F9F9F9] rounded-[10px] pl-[8px] pr-[8px]  gap-[10px] flex">
      <IconBox />
      <Details
        company={company}
        location={location}
        title={title}
        timeAgo={timeAgo}
      />
    </div>
  );
};

const IconBox = () => {
  return (
    <div className="flex items-center justify-center w-[38px] h-[38px]">
      <Image
        src={"/figma.png"}
        alt="company logo"
        width={38}
        height={38}
        className="rounded-full object-cover"
      />
    </div>
  );
};

const Details = ({
  company,
  location,
  title,
  timeAgo,
}: {
  company: string;
  location: string;
  title: string;
  timeAgo: string;
}) => {
  return (
    <div className="w-[217px] h-[37px] gap-[4px] flex flex-col">
      <TextElement6 text={title} textColor="#333333" fontWeight="font-medium" />
      <TextElement6
        text={`${company}, ${location} - ${timeAgo}`}
        textColor="#8F8F8F"
        fontWeight="font-normal"
      />
    </div>
  );
};

const TextElement6 = ({
  text,
  textColor,
  fontWeight,
}: {
  text: string;
  textColor: string;
  fontWeight: string;
}) => {
  return (
    <div className="w-[217px] h-[18px] flex items-center">
      <p
        style={{ color: textColor }}
        className={`font-inter ${fontWeight} text-[12px] leading-[15px] tracking-[0%]`}
      >
        {text}
      </p>
    </div>
  );
};
