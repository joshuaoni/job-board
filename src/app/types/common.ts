export interface LoadingSpinnerProps {
  size?: "small" | "medium" | "large";
  className?: string;
}
