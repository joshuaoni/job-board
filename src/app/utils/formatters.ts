export const formatJobType = (jobType: string): string => {
  const formattedTypes: { [key: string]: string } = {
    full_time: "Full time",
    part_time: "Part time",
    hybrid: "Hybrid",
    internship: "Internship",
  };

  return formattedTypes[jobType.toLowerCase()] || jobType;
};
