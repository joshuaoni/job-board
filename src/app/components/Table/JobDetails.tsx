import Image from "next/image";
import { TitleDetailsProps } from "@/app/types/job";

const TextElement = ({
  text,
  textColor,
  fontWeight,
}: {
  text: string;
  textColor: string;
  fontWeight: string;
}) => {
  return (
    <div className="w-[217px] h-[18px] flex items-center">
      <p
        style={{ color: textColor }}
        className={`font-inter ${fontWeight} text-[12px] leading-[15px] tracking-[0%] truncate`}
      >
        {text}
      </p>
    </div>
  );
};

const Details = ({ company, location, title, timeAgo }: TitleDetailsProps) => {
  return (
    <div className="w-[217px] h-[37px] gap-[4px] flex flex-col">
      <TextElement text={title} textColor="#333333" fontWeight="font-medium" />
      <TextElement
        text={`${company}, ${location} - ${timeAgo}`}
        textColor="#8F8F8F"
        fontWeight="font-normal"
      />
    </div>
  );
};

export const JobDetails = ({
  company,
  location,
  title,
  timeAgo,
}: TitleDetailsProps) => {
  return (
    <div className="h-[51px] rounded-[10px] p-[4px] pl-0 gap-[10px] flex items-center">
      <div className="w-full h-full bg-[#F9F9F9] rounded-[10px] pl-[8px] pr-[8px] gap-[10px] flex">
        <div className="flex items-center justify-center w-[38px] h-[38px]">
          <Image
            src="/figma.png"
            alt={`${company} logo`}
            width={38}
            height={38}
            className="rounded-full object-cover"
          />
        </div>
        <Details
          company={company}
          location={location}
          title={title}
          timeAgo={timeAgo}
        />
      </div>
    </div>
  );
};
