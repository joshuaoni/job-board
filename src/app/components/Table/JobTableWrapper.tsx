"use client";

import { useEffect, useState } from "react";
import { Table } from "./Table";
import { getJobs } from "@/app/services/jobService";
import { PaginationContainer } from "../Pagination/Pagination";

export default function JobTableWrapper({
  searchParams,
}: {
  searchParams: any;
}) {
  const [jobs, setJobs] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const page = Number(searchParams.page) || 1;
  const pageSize = 10;

  useEffect(() => {
    const fetchJobs = async () => {
      setIsLoading(true);
      const jobsData = await getJobs({
        search_term: searchParams.search_term || "",
        job_type: searchParams.job_type?.toLowerCase() || "full_time",
        location: searchParams.location?.toLowerCase() || "",
        skills: searchParams.skills
          ? searchParams.skills.split(",").map((s: string) => s.trim())
          : [],
      });
      setJobs(jobsData);
      setIsLoading(false);
    };

    fetchJobs();
  }, [searchParams]); // Re-fetch when searchParams change

  // Pagination Calculation
  const totalItems = jobs.length;
  const totalPages = Math.ceil(totalItems / pageSize);
  const startIndex = (page - 1) * pageSize;
  const endIndex = startIndex + pageSize;
  const currentPageItems = jobs.slice(startIndex, endIndex);

  return (
    <>
      <Table data={currentPageItems} isLoading={isLoading} />
      <PaginationContainer
        currentPage={page}
        totalPages={totalPages}
        hasNextPage={endIndex < totalItems}
        hasPrevPage={page > 1}
      />
    </>
  );
}
