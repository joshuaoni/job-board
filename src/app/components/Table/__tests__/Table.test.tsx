import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import { Table } from "../Table";
import { formatJobType } from "@/app/utils/formatters";
import { Job } from "@/app/types/job";

// Mock next/navigation
const mockPush = jest.fn();
jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

// Mock formatJobType
jest.mock("@/app/utils/formatters", () => ({
  formatJobType: jest.fn((type) => `Formatted ${type}`),
}));

describe("Table", () => {
  const mockJobs: Job[] = [
    {
      id: "1",
      job_title: "Software Engineer",
      company_name: "Tech Corp",
      job_type: "full_time",
      job_location_name: "New York, NY",
      years_of_experience_required: "3-5 years",
      salary_range_min: 80000,
      salary_range_max: 120000,
      salary_currency: "USD",
      created_at: new Date("2024-01-15").toISOString(),
      company_logo: null,
      required_skills: "React, TypeScript",
      languages: "English",
      tags: "Remote",
      company_description: "A tech company",
      job_description: "Software engineering role",
      educational_requirements: "Bachelor's degree",
      additional_benefits: "Health insurance",
      company_website: null,
    },
    {
      id: "2",
      job_title: "Product Manager",
      company_name: "Product Co",
      job_type: "part_time",
      job_location_name: "Remote",
      years_of_experience_required: "5+ years",
      salary_range_min: 90000,
      salary_range_max: 150000,
      salary_currency: "USD",
      created_at: new Date("2024-01-14").toISOString(),
      company_logo: null,
      required_skills: "Product Management, Agile",
      languages: "English, Spanish",
      tags: "Tech, Product",
      company_description: "A product company",
      job_description: "Product management role",
      educational_requirements: "Master's degree",
      additional_benefits: "Stock options",
      company_website: null,
    },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render table headers", () => {
    render(<Table data={mockJobs} />);

    expect(screen.getByText("Job Title")).toBeInTheDocument();
    expect(screen.getByText("Job Type")).toBeInTheDocument();
    expect(screen.getByText("Skills")).toBeInTheDocument();
    expect(screen.getByText("Languages")).toBeInTheDocument();
    expect(screen.getByText("Tags")).toBeInTheDocument();
  });

  it("should render job rows", () => {
    render(<Table data={mockJobs} />);

    expect(screen.getAllByText("Software Engineer")[0]).toBeInTheDocument();
    expect(screen.getByText("Tech Corp")).toBeInTheDocument();
    expect(screen.getByText("Formatted full_time")).toBeInTheDocument();
    expect(screen.getByText("New York, NY")).toBeInTheDocument();

    expect(screen.getAllByText("Product Manager")[0]).toBeInTheDocument();
    expect(screen.getByText("Product Co")).toBeInTheDocument();
    expect(screen.getByText("Formatted part_time")).toBeInTheDocument();
    expect(screen.getAllByText("Remote")[0]).toBeInTheDocument();
  });

  it("should format job types", () => {
    render(<Table data={mockJobs} />);

    expect(formatJobType).toHaveBeenCalledWith("full_time");
    expect(formatJobType).toHaveBeenCalledWith("part_time");
  });

  it("should show loading skeleton when loading", () => {
    render(<Table data={[]} isLoading={true} />);

    const skeleton = screen.getByTestId("table-skeleton");
    expect(skeleton).toBeInTheDocument();
  });

  it("should show no jobs message when empty", () => {
    render(<Table data={[]} />);

    expect(
      screen.getByText("No jobs found matching your criteria")
    ).toBeInTheDocument();
  });

  it("should handle row click", () => {
    render(<Table data={mockJobs} />);

    const firstRow = screen.getAllByTestId("table-row")[0];
    fireEvent.click(firstRow);

    expect(mockPush).toHaveBeenCalledWith("/jobs/1");
  });

  it("should be responsive", () => {
    render(<Table data={mockJobs} />);

    const mobileView = screen.getAllByTestId("table-row-mobile")[0];
    expect(mobileView).toHaveClass("md:hidden");

    const desktopView = screen.getAllByTestId("table-row-desktop")[0];
    expect(desktopView).toHaveClass("hidden", "md:grid");
  });
});
