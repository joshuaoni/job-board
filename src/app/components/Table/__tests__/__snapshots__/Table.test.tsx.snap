// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot when empty 1`] = `
<div>
  <div
    class="w-full my-4 animate-[fadeIn_0.3s_ease-in-out]"
  >
    <div
      class="hidden md:grid w-full h-[39.292px] rounded-[7.76px] pt-[11.65px] pb-[11.65px] bg-[#D6D6D6] gap-4 grid-cols-[2fr_1fr_1fr_1fr_1fr] items-center mb-4"
    >
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Job Title
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Job Type
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Skills
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Languages
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Tags
        </p>
      </div>
    </div>
    <div
      class="w-full py-8 text-center text-gray-500 animate-[fadeIn_0.3s_ease-in-out]"
    >
      No jobs found matching your criteria
    </div>
  </div>
</div>
`;

exports[`Table should match snapshot when loading 1`] = `
<div>
  <div
    class="w-full my-4"
    data-testid="table-skeleton"
  >
    <div
      class="w-full h-[39.292px] rounded-[7.76px] bg-gray-200 mb-4 hidden md:block"
    />
    <div
      class="w-full border-b border-gray-100 animate-pulse"
    >
      <div
        class="md:hidden p-4"
      >
        <div
          class="space-y-3"
        >
          <div
            class="w-3/4 h-5 bg-gray-100 rounded"
          />
          <div
            class="w-1/2 h-4 bg-gray-100 rounded"
          />
          <div
            class="flex gap-2"
          >
            <div
              class="w-20 h-4 bg-gray-100 rounded"
            />
            <div
              class="w-24 h-4 bg-gray-100 rounded"
            />
          </div>
        </div>
      </div>
      <div
        class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center"
      >
        <div
          class="h-full rounded-[10px] p-1 pl-0"
        >
          <div
            class="h-full w-full bg-gray-100 rounded-[6px]"
          />
        </div>
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
      </div>
    </div>
    <div
      class="w-full border-b border-gray-100 animate-pulse"
    >
      <div
        class="md:hidden p-4"
      >
        <div
          class="space-y-3"
        >
          <div
            class="w-3/4 h-5 bg-gray-100 rounded"
          />
          <div
            class="w-1/2 h-4 bg-gray-100 rounded"
          />
          <div
            class="flex gap-2"
          >
            <div
              class="w-20 h-4 bg-gray-100 rounded"
            />
            <div
              class="w-24 h-4 bg-gray-100 rounded"
            />
          </div>
        </div>
      </div>
      <div
        class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center"
      >
        <div
          class="h-full rounded-[10px] p-1 pl-0"
        >
          <div
            class="h-full w-full bg-gray-100 rounded-[6px]"
          />
        </div>
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
      </div>
    </div>
    <div
      class="w-full border-b border-gray-100 animate-pulse"
    >
      <div
        class="md:hidden p-4"
      >
        <div
          class="space-y-3"
        >
          <div
            class="w-3/4 h-5 bg-gray-100 rounded"
          />
          <div
            class="w-1/2 h-4 bg-gray-100 rounded"
          />
          <div
            class="flex gap-2"
          >
            <div
              class="w-20 h-4 bg-gray-100 rounded"
            />
            <div
              class="w-24 h-4 bg-gray-100 rounded"
            />
          </div>
        </div>
      </div>
      <div
        class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center"
      >
        <div
          class="h-full rounded-[10px] p-1 pl-0"
        >
          <div
            class="h-full w-full bg-gray-100 rounded-[6px]"
          />
        </div>
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
      </div>
    </div>
    <div
      class="w-full border-b border-gray-100 animate-pulse"
    >
      <div
        class="md:hidden p-4"
      >
        <div
          class="space-y-3"
        >
          <div
            class="w-3/4 h-5 bg-gray-100 rounded"
          />
          <div
            class="w-1/2 h-4 bg-gray-100 rounded"
          />
          <div
            class="flex gap-2"
          >
            <div
              class="w-20 h-4 bg-gray-100 rounded"
            />
            <div
              class="w-24 h-4 bg-gray-100 rounded"
            />
          </div>
        </div>
      </div>
      <div
        class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center"
      >
        <div
          class="h-full rounded-[10px] p-1 pl-0"
        >
          <div
            class="h-full w-full bg-gray-100 rounded-[6px]"
          />
        </div>
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
      </div>
    </div>
    <div
      class="w-full border-b border-gray-100 animate-pulse"
    >
      <div
        class="md:hidden p-4"
      >
        <div
          class="space-y-3"
        >
          <div
            class="w-3/4 h-5 bg-gray-100 rounded"
          />
          <div
            class="w-1/2 h-4 bg-gray-100 rounded"
          />
          <div
            class="flex gap-2"
          >
            <div
              class="w-20 h-4 bg-gray-100 rounded"
            />
            <div
              class="w-24 h-4 bg-gray-100 rounded"
            />
          </div>
        </div>
      </div>
      <div
        class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center"
      >
        <div
          class="h-full rounded-[10px] p-1 pl-0"
        >
          <div
            class="h-full w-full bg-gray-100 rounded-[6px]"
          />
        </div>
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
        <div
          class="h-5 bg-gray-100 rounded"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Table should match snapshot with data 1`] = `
<div>
  <div
    class="w-full my-4 animate-[fadeIn_0.3s_ease-in-out]"
  >
    <div
      class="hidden md:grid w-full h-[39.292px] rounded-[7.76px] pt-[11.65px] pb-[11.65px] bg-[#D6D6D6] gap-4 grid-cols-[2fr_1fr_1fr_1fr_1fr] items-center mb-4"
    >
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Job Title
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Job Type
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Skills
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Languages
        </p>
      </div>
      <div
        class="h-[16px] first:pl-4 text-[#898989] flex items-center justify-start"
      >
        <p
          class="font-inter font-bold text-[13.59px] leading-[100%] tracking-[5%]"
        >
          Tags
        </p>
      </div>
    </div>
    <div
      class="w-full divide-y divide-gray-100"
    >
      <div
        class="w-full hover:bg-gray-50 cursor-pointer transition-all duration-200 animate-[slideUp_0.3s_ease-out]"
        data-testid="table-row"
        style="animation-delay: 0s;"
      >
        <div
          class="md:hidden p-4"
          data-testid="table-row-mobile"
        >
          <div
            class="space-y-2"
          >
            <h3
              class="font-medium text-base text-black"
            >
              Software Engineer
            </h3>
            <p
              class="text-sm text-gray-600"
            >
              Tech Corp
            </p>
            <div
              class="flex flex-wrap gap-2 text-xs text-gray-500"
            >
              <span>
                New York, NY
              </span>
              <span>
                •
              </span>
              <span>
                1 year ago
              </span>
            </div>
            <div
              class="flex flex-wrap gap-2 mt-2"
            >
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                full_time
              </span>
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                React
              </span>
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                TypeScript
              </span>
            </div>
          </div>
        </div>
        <div
          class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center hover:scale-[1.01]"
          data-testid="table-row-desktop"
        >
          <div
            class="h-[51px] rounded-[10px] p-[4px] pl-0 gap-[10px] flex items-center"
          >
            <div
              class="w-full h-full bg-[#F9F9F9] rounded-[10px] pl-[8px] pr-[8px] gap-[10px] flex"
            >
              <div
                class="flex items-center justify-center w-[38px] h-[38px]"
              >
                <img
                  alt="Tech Corp logo"
                  class="rounded-full object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="38"
                  loading="lazy"
                  src="/_next/image?url=%2Ffigma.png&w=96&q=75"
                  srcset="/_next/image?url=%2Ffigma.png&w=48&q=75 1x, /_next/image?url=%2Ffigma.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="38"
                />
              </div>
              <div
                class="w-[217px] h-[37px] gap-[4px] flex flex-col"
              >
                <div
                  class="w-[217px] h-[18px] flex items-center"
                >
                  <p
                    class="font-inter font-medium text-[12px] leading-[15px] tracking-[0%] truncate"
                    style="color: rgb(51, 51, 51);"
                  >
                    Software Engineer
                  </p>
                </div>
                <div
                  class="w-[217px] h-[18px] flex items-center"
                >
                  <p
                    class="font-inter font-normal text-[12px] leading-[15px] tracking-[0%] truncate"
                    style="color: rgb(143, 143, 143);"
                  >
                    Tech Corp, New York, NY - 1 year ago
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full h-[19px] flex items-center justify-start"
          >
            <p
              class="font-inter font-normal text-[16px] leading-[100%] tracking-[5%] text-black"
            >
              Formatted full_time
            </p>
          </div>
          <div
            class="w-full flex justify-start items-center gap-2 overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="React, TypeScript"
              >
                React
              </span>
              <span
                class="ml-1 px-2 py-0.5 rounded-full text-xs font-medium bg-gray-100 text-gray-600"
                title="TypeScript"
              >
                +
                1
              </span>
            </div>
          </div>
          <div
            class="w-full flex justify-start items-center overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="English"
              >
                English
              </span>
            </div>
          </div>
          <div
            class="w-full overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="Remote"
              >
                Remote
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full hover:bg-gray-50 cursor-pointer transition-all duration-200 animate-[slideUp_0.3s_ease-out]"
        data-testid="table-row"
        style="animation-delay: 0.05s;"
      >
        <div
          class="md:hidden p-4"
          data-testid="table-row-mobile"
        >
          <div
            class="space-y-2"
          >
            <h3
              class="font-medium text-base text-black"
            >
              Product Manager
            </h3>
            <p
              class="text-sm text-gray-600"
            >
              Product Co
            </p>
            <div
              class="flex flex-wrap gap-2 text-xs text-gray-500"
            >
              <span>
                Remote
              </span>
              <span>
                •
              </span>
              <span>
                1 year ago
              </span>
            </div>
            <div
              class="flex flex-wrap gap-2 mt-2"
            >
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                part_time
              </span>
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                Product Management
              </span>
              <span
                class="px-2 py-1 bg-gray-100 rounded-full text-xs"
              >
                Agile
              </span>
            </div>
          </div>
        </div>
        <div
          class="hidden md:grid h-[51px] grid-cols-[2fr_1fr_1fr_1fr_1fr] gap-4 items-center hover:scale-[1.01]"
          data-testid="table-row-desktop"
        >
          <div
            class="h-[51px] rounded-[10px] p-[4px] pl-0 gap-[10px] flex items-center"
          >
            <div
              class="w-full h-full bg-[#F9F9F9] rounded-[10px] pl-[8px] pr-[8px] gap-[10px] flex"
            >
              <div
                class="flex items-center justify-center w-[38px] h-[38px]"
              >
                <img
                  alt="Product Co logo"
                  class="rounded-full object-cover"
                  data-nimg="1"
                  decoding="async"
                  height="38"
                  loading="lazy"
                  src="/_next/image?url=%2Ffigma.png&w=96&q=75"
                  srcset="/_next/image?url=%2Ffigma.png&w=48&q=75 1x, /_next/image?url=%2Ffigma.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="38"
                />
              </div>
              <div
                class="w-[217px] h-[37px] gap-[4px] flex flex-col"
              >
                <div
                  class="w-[217px] h-[18px] flex items-center"
                >
                  <p
                    class="font-inter font-medium text-[12px] leading-[15px] tracking-[0%] truncate"
                    style="color: rgb(51, 51, 51);"
                  >
                    Product Manager
                  </p>
                </div>
                <div
                  class="w-[217px] h-[18px] flex items-center"
                >
                  <p
                    class="font-inter font-normal text-[12px] leading-[15px] tracking-[0%] truncate"
                    style="color: rgb(143, 143, 143);"
                  >
                    Product Co, Remote - 1 year ago
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full h-[19px] flex items-center justify-start"
          >
            <p
              class="font-inter font-normal text-[16px] leading-[100%] tracking-[5%] text-black"
            >
              Formatted part_time
            </p>
          </div>
          <div
            class="w-full flex justify-start items-center gap-2 overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="Product Management, Agile"
              >
                Product Management
              </span>
              <span
                class="ml-1 px-2 py-0.5 rounded-full text-xs font-medium bg-gray-100 text-gray-600"
                title="Agile"
              >
                +
                1
              </span>
            </div>
          </div>
          <div
            class="w-full flex justify-start items-center overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="English, Spanish"
              >
                English
              </span>
              <span
                class="ml-1 px-2 py-0.5 rounded-full text-xs font-medium bg-gray-100 text-gray-600"
                title="Spanish"
              >
                +
                1
              </span>
            </div>
          </div>
          <div
            class="w-full overflow-hidden"
          >
            <div
              class="flex justify-start items-center overflow-hidden"
            >
              <span
                class="text-sm text-black truncate"
                title="Tech, Product"
              >
                Tech
              </span>
              <span
                class="ml-1 px-2 py-0.5 rounded-full text-xs font-medium bg-blue-50 text-blue-600"
                title="Product"
              >
                +
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
